src/main/java/com/stripe/net/APIResource.java

```diff
@@ -130,4 +131,27 @@ public static String urlEncode(String str) throws UnsupportedEncodingException {
 		return APIResource.stripeResponseGetter.request(method, url, params, clazz,
 				APIResource.RequestType.NORMAL, options);
 	}
+
+    /**
+     * Similar to #request, but specific for use with collection types that
+     * come from the API (i.e. lists of resources).
+     *
+     * Collections need a little extra work because we need to plumb request
+     * options and params through so that we can iterate to the next page if
+     * necessary.
+     */
+    protected static <T extends StripeCollectionInterface> T requestCollection(
```

Comment:
@jimdanz Glad like you like it! It took a fair bit of refactoring, but agreed, the result isn't the worst :)